kernel/typeops.ml

```diff
@@ -213,9 +213,20 @@ let type_of_apply env func funt argsv argstv =
   apply_rec 0 (inject funt)
 
 (* Type of primitive constructs *)
-let type_of_prim_type _env = function
-  | CPrimitives.PT_int63 -> Constr.mkSet
-  | CPrimitives.PT_float64 -> Constr.mkSet
+let type_of_prim_type _env u (type a) (prim : a CPrimitives.prim_type) = match prim with
+  | CPrimitives.PT_int63 ->
+    assert (Univ.Instance.is_empty u);
+    Constr.mkSet
+  | CPrimitives.PT_float64 ->
+    assert (Univ.Instance.is_empty u);
+    Constr.mkSet
+  | CPrimitives.PT_array ->
+    begin match Univ.Instance.to_array u with
+    | [|u|] ->
+      let ty = Constr.mkType (Univ.Universe.make u) in
+      Constr.mkProd(Context.anonR, ty , ty)
```

Comment:
Can you explain this rule?
It looks like it imitates `Inductive array (A : Type@{u}) : Type@{u} := ..` which looks weird to me.

Reply:
(parameters do not count for the level of the inductive, right?)

Reply:
What did you expect? I don't see what's weird about it.

Reply:
Hum, nevermind I got confused. I was expecting Coq would compute a `v >= u` which is not the case.